x-pack/plugins/ingest_manager/server/routes/data_streams/handlers.ts

```diff
@@ -99,14 +104,31 @@ export const getListHandler: RequestHandler = async (context, request, response)
         package: { buckets: packageBuckets },
         last_activity: { value_as_string: lastActivity },
       } = result;
+
+      const pkg = packageBuckets.length ? packageBuckets[0].key : '';
+      const pkgSavedObject = packageSavedObjects.saved_objects.filter(p => p.id === pkg);
+
+      // if
+      // - the datastream is associated with a package
+      // - and the package has been installed through EPM
+      // - and we didn't pick the metadata in an earlier iteration of this map()
+      if (pkg !== '' && pkgSavedObject.length > 0 && !packageMetadata[pkg]) {
+        packageMetadata[pkg] = {
+          version: pkgSavedObject[0].attributes?.version || '',
+          dashboards:
+            pkgSavedObject[0].attributes?.installed?.filter(o => o.type === 'dashboard') || [],
+        };
+      }
       return {
         index: indexName,
         dataset: datasetBuckets.length ? datasetBuckets[0].key : '',
         namespace: namespaceBuckets.length ? namespaceBuckets[0].key : '',
         type: typeBuckets.length ? typeBuckets[0].key : '',
-        package: packageBuckets.length ? packageBuckets[0].key : '',
+        package: pkg,
+        package_version: packageMetadata[pkg] ? packageMetadata[pkg].version : '',
```

Comment:
After testing locally, I realized we still display the package id instead of the nice name (`system` vs `System`). Now that we have the package saved objects here, what do you think about restructuring `DataStream` type to include package name?:
```
package: {
  id: string;
  name: string;
  version: string;
}
```

Not a blocker for FF tho :)

Reply:
Yes, we should do that for beta1. We'll need to add the `title` from the registry package information to our `epm-packages` saved object type, and I'm reluctant to do that right now, so this is https://github.com/elastic/kibana/issues/65230 now.